cmake/Modules/SYCL.cmake

```diff
@@ -24,11 +24,54 @@
 # **************************************************************************/
 include(CheckCXXCompilerFlag)
 include(ConfigureSYCLBLAS)
-check_cxx_compiler_flag("-fsycl" is_dpcpp)
-if(NOT is_dpcpp)
-  set (is_computecpp ON)
+
+# set hipSYCL defaults if not set before the package is searched
+# as that would fail otherwise
+if(NOT HISPYCL_TARGETS AND NOT ENV{HIPSYCL_TARGETS})
```

Comment:
can you move it inside the `elseif(is_hipsycl)` please, since this configuration will be only used when HIPSYCL is used.